conda_build/build.py

```diff
@@ -519,61 +516,55 @@ def is_no_link(no_link, short_path):
-    ensure_list(files)
-    target_short_path_inode = os.stat(join(prefix, target_short_path)).st_ino
+    target_short_path_inode = os.lstat(join(prefix, target_short_path)).st_ino
     hardlinked_files = [sp for sp in files
-                        if os.stat(join(prefix, sp)).st_ino == target_short_path_inode]
+                        if os.lstat(join(prefix, sp)).st_ino == target_short_path_inode]
     return sorted(hardlinked_files)
 
 
-def file_type(path):
-    if isdir(path):
-        return FileType.directory
-    elif islink(path):
-        return FileType.softlink
-    return FileType.hardlink
+def path_type(path):
+    if islink(path):
+        return PathType.softlink
+    return PathType.hardlink
 
 
-def build_info_files_json(m, prefix, files, files_with_prefix):
+def build_info_files_json_v1(m, prefix, files, files_with_prefix):
     no_link = m.get_value('build/no_link')
     files_json = []
-    for fi in files:
+    for fi in sorted(files):
         prefix_placeholder, file_mode = has_prefix(fi, files_with_prefix)
         path = os.path.join(prefix, fi)
         file_info = {
-            "path": get_short_path(m, fi),
+            "_path": get_short_path(m, fi),
             "sha256": sha256_checksum(path),
             "size_in_bytes": os.path.getsize(path),
-            "file_type": getattr(file_type(path), "name"),
+            "path_type": path_type(path),
         }
         no_link = is_no_link(no_link, fi)
         if no_link:
             file_info["no_link"] = no_link
         if prefix_placeholder and file_mode:
             file_info["prefix_placeholder"] = prefix_placeholder
             file_info["file_mode"] = file_mode
-        if file_info.get("file_type") == "hardlink" and os.stat(join(prefix, fi)).st_nlink > 1:
+        if file_info.get("path_type") == PathType.hardlink and CrossPlatformStLink.st_nlink(
+                join(prefix, fi)) > 1:
             inode_paths = get_inode_paths(files, fi, prefix)
             file_info["inode_paths"] = inode_paths
         files_json.append(file_info)
     return files_json
 
 
-def get_files_version():
-    return 1
-
-
-def create_info_files_json(m, info_dir, prefix, files, files_with_prefix):
-    files_json_fields = ["path", "sha256", "size_in_bytes", "file_type", "file_mode",
-                         "prefix_placeholder", "no_link", "inode_first_path"]
-    files_json_files = build_info_files_json(m, prefix, files, files_with_prefix)
+def create_info_files_json_v1(m, info_dir, prefix, files, files_with_prefix):
+    # fields: "_path", "sha256", "size_in_bytes", "path_type", "file_mode",
+    #         "prefix_placeholder", "no_link", "inode_paths"
+    files_json_files = build_info_files_json_v1(m, prefix, files, files_with_prefix)
     files_json_info = {
-        "version": get_files_version(),
-        "fields": files_json_fields,
-        "files": files_json_files,
+        "paths_version": 1,
+        "paths": files_json_files,
```

Comment:
For overall compatibility with linked data json files in `conda-meta/`, rename this `paths` key back to `files.